### PR TITLE
Scraper & processor

### PR DESCRIPTION
**Pull request by ashersnavely**
Just overall process/syntax review sometime tomorrow.

### CODE REVIEW

**Review by johnsimmons2 — COMMENTED** [review]
There is a lot of redundant classes here, StackOverflow shows up more than once as does ReleaseHeap. Did you separate them out into other files but not delete their original occurrences?

**Comment by johnsimmons2** [comment]
> There is a lot of redundant classes here, StackOverflow shows up more than once as does ReleaseHeap. Did you separate them out into other files but not delete their original occurrences?

I see /scraping and /scraper, is there a difference? Do these classes need 2 definitions

File: stackoversight/scraper/release_heap.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Need a heap ofc
+import heapq
+# For release timers
+import threading
+# For the release queue
+from queue import Queue
+
+
+class ReleaseHeap(object):
```
Comment (johnsimmons2) [inline stackoversight/scraper/release_heap.py]: I notice there is ReleaseHeap(object) and ReleaseHeap(queue) (I do not know if doing (object) is standard notation or something, but it is redundant and not needed for Python 3) do both classes need to exist? If so consider changing their names to show explicit differences

File: stackoversight/scraping/site/Site.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            else:
+                content = {parse_tree.find(tag).get_text()}
+        except:
+            # sometimes there wont be any matching tags
+            return None
+
+        return content
+
+
+class StackOverflow(Site):
```
Comment (johnsimmons2) [inline stackoversight/scraping/site/Site.py]: 2nd occurrence of StackOverflow(Site) class definition

**Comment by ashersnavely** [comment]
Accidentally left in old files, thanks for catching that.

They should be removed now!

File: stackoversight/scraper/release_heap.py (hunk 1 above)
Comment (johnsimmons2) [reply stackoversight/scraper/release_heap.py]: This is resolved with latest push

File: stackoversight/scraper/scraper.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    print(question)
+    soup = site.get_soup(question, pause=True)
+
+    for code in site.get_code(soup):
+        print(code)
+
+    for text in site.get_text(soup):
+        print(text)
+
+print(site.balancer.heap)
```
Comment (johnsimmons2) [inline stackoversight/scraper/scraper.py]: When does this script get called or is this just for testing?

File: stackoversight/scraper/release_heap.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    """
+    def __init__(self, elem_list: list, time_sec: int):
+        # init the heap and queue and assign variables
+        heapq.heapify(elem_list)
+        self.heap = elem_list
+        self.time_sec = time_sec
+        self.release_queue = Queue()
+
+        # quick access to pair by it's value
+        self.hash_map = dict([(pair[1], pair) for pair in self.heap])
```
Comment (johnsimmons2) [inline stackoversight/scraper/release_heap.py]: I remember this line causing trouble earlier due to tuples being immutable, is this workable now?

**Review by johnsimmons2 — COMMENTED** [review]
2 questions/comments but otherwise it is good to merge

File: stackoversight/scraper/scraper.py (hunk 3 above)
Comment (ashersnavely) [reply stackoversight/scraper/scraper.py]: Just for testing!

File: stackoversight/scraper/release_heap.py (hunk 4 above)
Comment (ashersnavely) [reply stackoversight/scraper/release_heap.py]: Yes, there is actually a class called "mutabletuple" and all i did was replace the non-mutable tuples with those.
Made sure to leave a comment about the type requirement before the ReleaseHeap init method.

Comment (johnsimmons2) [reply stackoversight/scraper/release_heap.py]: I saw that elsewhere, but why is there no mutabletuple here? @ashersnavely just curious

**Comment by johnsimmons2** [comment]
@ashersnavely is this closer to being able to pull and close?

or do you have a different branch you're working on, just wondering so I can merge

**Pull request closed without merging**